libsolidity/ast/AST.cpp

```diff
@@ -514,20 +514,16 @@ ModifierDefinition const& ModifierDefinition::resolveVirtual(
 	solAssert(_searchStart == nullptr, "Used super in connection with modifiers.");
 
 	// If we are not doing super-lookup and the modifier is not virtual, we can stop here.
-	if (_searchStart == nullptr && !virtualSemantics())
+	if (!virtualSemantics())
 		return *this;
 
 	solAssert(!dynamic_cast<ContractDefinition const&>(*scope()).isLibrary(), "");
 
 	for (ContractDefinition const* c: _mostDerivedContract.annotation().linearizedBaseContracts)
-	{
-		if (_searchStart != nullptr && c != _searchStart)
```

Comment:
I am wondering what the intention of this check was prior to this PR.

- `_searchStart` will always be nullptr here (because of previous assert plus following assignment)
- `c != _searchStart` is essentially `c != nullptr` (this does not make sense)

Reply:
I think the assert was added after this code was written.